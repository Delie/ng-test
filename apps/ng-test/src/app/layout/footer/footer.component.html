{{ 'Language' | translate }}: <a (click)="setLanguage('en')">EN</a> | <a (click)="setLanguage('fr')">FR</a>
