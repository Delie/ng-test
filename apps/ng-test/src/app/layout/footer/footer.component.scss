:host {
  float: left;
  width: 100%;
  padding: 10px;
  background: #555;
  color: #fff;
}
