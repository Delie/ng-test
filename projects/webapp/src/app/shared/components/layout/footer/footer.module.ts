import { NgModule } from '@angular/core';
import { FooterComponent } from './footer.component';
import { TranslateModule } from '@ngx-translate/core';

@NgModule({
  declarations: [FooterComponent],
  imports: [TranslateModule],
  exports: [FooterComponent]
})
export class FooterModule {}
