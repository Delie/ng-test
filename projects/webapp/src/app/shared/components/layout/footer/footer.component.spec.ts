import { async } from '@angular/core/testing';
import { Shallow } from 'shallow-render';
import { FooterComponent } from './footer.component';
import { FooterModule } from './footer.module';
import { TranslateService, TranslateModule } from '@ngx-translate/core';

describe('FooterComponent', () => {
  let shallow: Shallow<FooterComponent>;
  beforeEach(async(() => {
    shallow = new Shallow(FooterComponent, FooterModule)
      .mock(TranslateService, {
        setDefaultLang: () => {},
        use: (): any => {}
      })
      .replaceModule(TranslateModule, TranslateModule.forRoot());
  }));

  it('should create the component', async () => {
    const { element } = await shallow.render();
    expect(element.nativeElement).toMatchSnapshot();
  });
});
