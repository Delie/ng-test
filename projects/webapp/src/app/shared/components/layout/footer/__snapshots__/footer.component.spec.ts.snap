// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterComponent should create the component 1`] = `
<app-footer>
  Language: 
  <a>
    EN
  </a>
   | 
  <a>
    FR
  </a>
</app-footer>
`;
